pkg/prometheus/statefulset.go

```diff
@@ -775,17 +775,31 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 			bindAddress = "127.0.0.1"
 		}
 
+		thanosArgs := []string{"sidecar",
+			fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, path.Clean(webRoutePrefix)),
+			fmt.Sprintf("--tsdb.path=%s", storageDir),
+			fmt.Sprintf("--grpc-address=%s:10901", bindAddress),
+			fmt.Sprintf("--http-address=%s:10902", bindAddress),
+		}
+
+		if p.Spec.Thanos.GRPCServerTLSConfig != nil {
```

Comment:
What are we configuring here?

Reply:
This is the config for the Thanos Sidecar attached to Prometheus.  It has the same tls options as ruler.